crates/holochain_cascade/src/authority/get_agent_activity_query/must_get_agent_activity.rs

```diff
@@ -83,32 +83,48 @@ pub fn filter_then_check(
     }
 }
 
-/// Find the filters sequence bounds.
+/// Find the filter's sequence bounds.
 fn find_bounds(
     txn: &Transaction,
     scratch: Option<&Scratch>,
     author: &AgentPubKey,
     filter: ChainFilter,
 ) -> StateQueryResult<Sequences> {
-    let mut statement = txn.prepare(ACTION_HASH_TO_SEQ)?;
+    let mut hash_statement = txn.prepare(ACTION_HASH_TO_SEQ)?;
 
     // Map an action hash to its sequence.
-    let get_seq = move |hash: &ActionHash| {
+    let get_seq_from_hash = |hash: &ActionHash| {
         if let Some(scratch) = scratch {
             if let Some(action) = scratch.actions().find(|a| a.action_address() == hash) {
                 return Ok(Some(action.action().action_seq()));
             }
         }
-        let result = statement
+        let result = hash_statement
                 .query_row(named_params! {":hash": hash, ":author": author, ":activity": ChainOpType::RegisterAgentActivity}, |row| {
                     row.get(0)
                 })
                 .optional()?;
         Ok(result)
     };
 
+    // Map a timestamp to its sequence.
+    let mut ts_statement = txn.prepare(ACTION_TS_TO_SEQ)?;
+    let get_seq_from_ts = |ts: Timestamp| {
+        if let Some(scratch) = scratch {
+            if let Some(action) = scratch.actions().find(|a| a.get_timestamp() == ts) {
```

Comment:
Timestamp equality? Intuitively, I'd expect a timestamp bound to include all the actions within the provided bound. The selected one for an action sequence would be the max or min depending on the direction of the equality. I can't make sense of an equality here.

Reply:
I've removed this part as I don't believe it actually makes sense to look in the scratch for old data. I had copied that over from how UntilHash was doing it.